backend/typescript/services/implementations/camperService.ts

```diff
@@ -897,6 +897,79 @@ class CamperService implements ICamperService {
     chargeId: string,
     camperIds: string[],
   ): Promise<void> {
+    try {
+      const campersWithChargeId: Array<Camper> = await MgCamper.find({
+        chargeId,
+      });
+      const campersToBeDeleted = campersWithChargeId.filter((camper) =>
+        camperIds.includes(camper.id),
+      );
+      const camperIdsToBeDeleted = campersToBeDeleted.map(
+        (camper) => camper.id,
+      );
+
+      if (!campersToBeDeleted.length) {
+        throw new Error(
+          `Campers with specified camperIds and charge ID ${chargeId} not found.`,
+        );
+      }
+
+      const refundAmountArray = await Promise.all(
+        camperIds.map(async (camperId) =>
+          this.cancelRegistrationSession(chargeId, [camperId]),
+        ),
+      );
+
+      const refundAmount = refundAmountArray.reduce((a, b) => a + b);
+
+      // refund before db deletion - a camper should not be deleted if the refund doesn't go through
+      await stripe.refunds.create({
+        charge: chargeId,
+        amount: refundAmount,
+      });
+
+      await this.deleteCampersById(camperIdsToBeDeleted);
+      await Promise.all(
+        campersToBeDeleted.map(async (camper) => {
+          const campSession: CampSession | null = await MgCampSession.findById(
+            camper.campSession,
+          );
+
+          if (!campSession) {
+            throw new Error(
+              `Camper's camp session with campId ${camper.campSession} not found.`,
+            );
+          }
+
+          const camp = await MgCamp.findById(campSession.camp);
+          if (!camp) {
+            throw new Error(
+              `Camper's camp with campId ${campSession.camp} not found.`,
+            );
+          }
+          emailService.sendAdminCamperCancellationNoticeEmail(
```

Comment:
Consider: Any chance we can batch these emails for less noise/spam? Not blocking PR merge though, just a thought